pkg/controller/serviceaccount/legacy_serviceaccount_token_cleaner.go

```diff
@@ -183,7 +186,29 @@ func (tc *LegacySATokenCleaner) evaluateSATokens(ctx context.Context) {
 			continue
 		}
 
-		logger.Info("Delete auto-generated service account token", "secret", klog.KRef(secret.Namespace, secret.Name), "creationTime", secret.CreationTimestamp, "lastUsed", lastUsed)
+		invalidSince := secret.Labels[InvalidSinceLabelKey]
+		// If the secret has not been labeled with invalid since date or the label value has invalid format, update the invalidSince label with the current date value.
+		_, err = time.Parse(dateFormat, invalidSince)
+		if err != nil {
+			invalidSince = now.Format(dateFormat)
+			logger.Info("Mark the auto-generated service account token as invalid", "invalidSince", invalidSince, "secret", klog.KRef(secret.Namespace, secret.Name))
+			patchContent, err := json.Marshal(applyv1.Secret(secret.Name, secret.Namespace).WithLabels(map[string]string{InvalidSinceLabelKey: invalidSince}))
+			if err != nil {
+				logger.Error(err, "Failed to marshal invalid since label")
+			} else {
+				if _, err := tc.client.CoreV1().Secrets(secret.Namespace).Patch(ctx, secret.Name, types.MergePatchType, patchContent, metav1.PatchOptions{}); err != nil {
+					logger.Error(err, "Failed to label legacy service account token secret with invalid since date")
+				}
+			}
+			continue
+		}
+
+		if invalidSince >= preserveUsedOnOrAfter {
+			logger.Info("Token has not been invalid for enough time", "minimal clean up period", tc.minimumSinceLastUsed, "invalid since", invalidSince)
```

Comment:
we could potentially have a lot of secrets hitting this and logging in steady state. either lower this verbosity significantly or drop the logging

Reply:
Removed the log.